engine/orchestration/src/com/cloud/vm/VirtualMachineManagerImpl.java

```diff
@@ -744,14 +744,17 @@ protected boolean checkWorkItems(final VMInstanceVO vm, final State state) throw
 
     protected <T extends VMInstanceVO> boolean changeState(final T vm, final Event event, final Long hostId, final ItWorkVO work, final Step step) throws NoTransitionException {
         // FIXME: We should do this better.
-        final Step previousStep = work.getStep();
-        _workDao.updateStep(work, step);
+        Step previousStep = null;
+        if (work != null) {
+            previousStep = work.getStep();
```

Comment:
Oh, and to your point earlier, getStep() generally shouldn't ever return a null I don't *think* , because the step column in the op_it_work table is marked not null.